react-components/src/components/CadModelContainer/CadModelContainer.tsx

```diff
@@ -101,6 +105,18 @@ export function CadModelContainer({
     viewer.removeModel(model);
     setModel(undefined);
   }
+
+  function applyLayersState(model: CogniteCadModel): void {
+    if (cadLayers === undefined) {
+      return;
+    }
+    const index = viewer.models.indexOf(model);
+    cadLayers.forEach((layer) => {
+      if (layer.revisionId === revisionId && layer.index === index) {
+        model.visible = layer.applied;
+      }
+    });
```

Comment:
Why are you using `forEach` here, can't you just use `array.find()` to get the value of the visibility for corresponding model? Also, we were talking about adding a check for index only in the case of duplicated models but not for all models.